Source/radiation/RadHydro.H

```diff
@@ -0,0 +1,298 @@
+// see Paper III, section 2.4
+
+constexpr int rk_order = 3;
+constexpr bool use_WENO = false;
+
+constexpr Real cfl_fspace = 0.5_rt;
+
+constexpr Real onethird = 1.0_rt/3.0_rt;
+constexpr Real twothirds = 2.0_rt/3.0_rt;
+constexpr Real onesixth = 1.0_rt/6.0_rt;
+
+// RK5
+constexpr Real B1 = 0.5_rt;
+constexpr Real B2 = 1.0_rt/16.0_rt;
+constexpr Real B3 = 0.5_rt;
+constexpr Real B4 = 9.0_rt/16.0_rt;
+constexpr Real B5 = 8.0_rt/7.0_rt;
+constexpr Real B6 = 7.0_rt/90.0_rt;
+
+constexpr Real C20 = 5.0_rt/8.0_rt;
+constexpr Real C21 = 3.0_rt/8.0_rt;
+
+constexpr Real C40 = 17.0_rt/8.0_rt;
+constexpr Real C41 = 9.0_rt/8.0_rt;
+constexpr Real C42 = -3.0_rt;
+constexpr Real C43 = 0.75_rt;
+
+constexpr Real C50 = -5.0_rt/21.0_rt;
+constexpr Real C51 = 2.0_rt/7.0_rt;
+constexpr Real C52 = 0.0_rt;
+constexpr Real C53 = 4.0_rt;
+constexpr Real C54 = -64.0_rt/21.0_rt;
+
+constexpr Real C60 = -8.0_rt/27.0_rt;
+constexpr Real C61 = -1.0_rt/5.0_rt;
+constexpr Real C62 = 32.0_rt/45.0_rt;
+constexpr Real C63 = -32.0_rt/45.0_rt;
+constexpr Real C64 = 32.0_rt/27.0_rt;
+constexpr Real C65 = 14.0_rt/45.0_rt;
+
+
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+Real
+MC(const Real r) {
+  return amrex::max(0.0_rt, amrex::min(2.0_rt * r, 0.5_rt * (1.0_rt + r), 2.0_rt));
+}
+
+
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+Real
+weno5(const Real vm2, const Real vm1, const Real v, const Real vp1, const Real vp2) {
+
+  constexpr Real epsw = 1.0e-6_rt;
+  constexpr Real b1 = 13.0_rt/12.0_rt;
+  constexpr Real b2 = 1.0_rt/6.0_rt;
+
+  Real djm1 = vm2 - 2.0_rt * vm1 + v;
+  Real ejm1 = vm2 - 4.0_rt * vm1 + 3.0_rt * v;
+  Real dj = vm1 - 2.0_rt * v + vp1;
+  Real ej = vm1 - vp1;
+  Real djp1 = v - 2.0_rt * vp1 + vp2;
+  Real ejp1 = 3.0_rt * v - 4.0_rt * vp1 + vp2;
+
+  Real dis0 = b1 * djm1 * djm1 + 0.25_rt * ejm1 * ejm1 + epsw;
+  Real dis1 = b1 * dj * dj + 0.25_rt * ej * ej + epsw;
+  Real dis2 = b1 * djp1 * djp1 + 0.25_rt * ejp1 * ejp1 + epsw;
+
+  Real q30 = 2.0_rt *vm2 - 7.0_rt * vm1 + 11.0_rt * v;
+  Real q31 = -vm1 + 5.0_rt * v + 2.0_rt * vp1;
+  Real q32 = 2.0_rt * v + 5.0_rt * vp1 - vp2;
+
+  Real d01 = dis0 / dis1;
+  Real d02 = dis0 / dis2;
+  Real a1ba0 = 6.0_rt * d01 * d01;
+  Real a2ba0 = 3.0_rt * d02 * d02;
+  Real w0 = 1.0_rt / (1.0_rt + a1ba0 + a2ba0);
+  Real w1 = a1ba0 * w0;
+  Real w2 = 1.0_rt - w0 - w1;
+
+  if (w0 <  1.0e-10_rt) {
+    w0 = 0.0_rt;
+  }
+  if (w1 < 1.0e-10_rt) {
+    w1 = 0.0_rt;
+  }
+  if (w2 < 1.0e-10_rt) {
+    w2 = 0.e0_rt;
+  }
+
+  return b2 * (w0 * q30 + w1 * q31 + w2 * q32);
+}
+
+
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void
+dudt(Real* u, Real* a, const Real* dx, const int n, Real* dudt_tmp) {
+
+  // compute the RHS for the g'th energy group out of n note, since
+  // this is advection in frequency space, we will need to assume some
+  // boundary conditions on the stencil used to reconstruct over nu
+
+  Real f[NGROUPS+1];
+
+  // we need 2 ghost cells on each end
+  Real ag[NGROUPS+4];
```

Comment:
Perhaps we should use Array1D here so we can retain the indexing from Fortran?

Reply:
done